### PR TITLE
chore: add doc 1.18.1 and adjust script for index based structure

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

- created index file with all links to the docs
- moved 1.18.0 doc to the `docs/1.18.0` folder
- added 1.18.1 doc in the `docs/1.18.1` folder and reflect it in the index file
- adjusted the script to 
    - move new docs to `docs/x.y.z` folder instead of under the `docs`
    - add new link of doc to the index file

## Definition of Done

Please consider all items and remove only if not applicable.

- [ ] Tests created/adjusted for your changes.
- [ ] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)

## Labels
We use labels to indicate the status of PRs. 
Please select `please review`, `please merge` or `don't merge` for your PR.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Documentation
+
+- [1.18.0](1.18.0)
+
+- [1.18.1](1.18.1)
```
Comment (marikaner) [inline docs/index.md]:
```suggestion
# API Documentation

- [Version 1.18.0](1.18.0)

- [Version 1.18.1](1.18.1)
```

**Comment by jjtang1985** [comment]
Please correct me, so you suggest:
- move all the other docs to `knowledge-base` folder under the root
- move the index file to `doc/api` folder

**Comment by marikaner** [comment]
> Please correct me, so you suggest:
> 
> * move all the other docs to `knowledge-base` folder under the root
> * move the index file to `doc/api` folder

Yes, that is my suggestion. Currently this is mixed and `release.md` is part of the official documentation: https://sap.github.io/cloud-sdk/release

**Comment by jjtang1985** [comment]
- move all the other docs to knowledge-base folder under the root
Sure, we should separate them
- move the index file to doc/api folder
Considering there might be some other topics in addition to api, it makes sense.
@jordan40 , please be aware this change, after this, the link of the docs will be:
https://sap.github.io/cloud-sdk/api instead of https://sap.github.io/cloud-sdk/

**Pull request merged**